update.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+# CHECK DISK FREE SPACE
+required_disk_free=200000
+disk_free=$(df -l | grep '/dev/root' | awk '{print $4}')
+echo "FREE DISK SPACE: $disk_free M"
+if [ "$disk_free" -lt "$required_disk_free" ]; then
+  echo "error > NOT ENOUGH FREE DISK SPACE (200M REQUIRED)"
+  exit
+fi
+
+# GET CURRENT VERSION
+version=$(cat /home/we/version.txt)
+echo "CURRENT VERSION: $version"
+
+# CHECK FOR USB DISK UPDATE FILES, COPY IF VALID
+disk_mount=$(lsblk -o mountpoint | grep media)
+if [ ! -f "$disk_mount" ]; then
+  echo "USB DISK PRESENT"
+  from_disk=$(ls $disk_mount | grep norns | sed 's/.norns//' | sort -r | head -1)
+  if [ "$from_disk" -gt "$version" ]; then
+    echo "> COPYING FROM USB DISK"
+    sudo cp -v $disk_mount/$from_disk.norns /home/we/update/
+  fi
+fi
+
+# CHECK FOR LOCAL UPDATE FILES
+local_update=$(ls /home/we/update | grep norns | sed 's/.norns//' | sort -r | head -1)
+echo "LOCAL UPDATE VERSION: $local_update"
+if [ -z "$local_update" ]; then
+  echo "error > NO UPDATES"
+  exit
+fi
+if [ "$version" -gt "$local_update" ]; then
+  echo "error > OLD UPDATE, NOT APPLYING"
+  echo "> REMOVING UPDATES"
+  rm -rf /home/we/update/*
```

Comment:
I can see how this would be helpful, but it makes me a bit nervous summarily deleting stuff that the user might have put there. Maybe we should instead give the user the commands to run to delete the old update files?

Reply:
Should be fine, right? I mean it's possible to write stuff there of course (via SSH/SCP), but who would do so and why?
I'd prefer to keep this as is, the updates are already problematic, relying on actions from users is only going to make it worse.